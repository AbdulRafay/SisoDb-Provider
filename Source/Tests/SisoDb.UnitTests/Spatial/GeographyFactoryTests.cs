using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using SisoDb.Spatial;

namespace SisoDb.UnitTests.Spatial
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GeographyFactoryTests : UnitTestBase
    {
        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderAndClosed(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderNotClosed(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderAndClosed(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderNotClosed(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_starting_south_to_north_and_many_to_west_and_back_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongSouthToNorthAndManyWestAndBackClosed(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Castle()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.Castle(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Star_that_is_wrong_from_the_beginning()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.StartThatDoesNotRenderWithoutFix(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_drawn_lower_right_counter_clockwise()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.TreasureIsland_drawn_lower_right_counter_clockwise(), SpatialReferenceId.Wsg84);

            Approvals.Verify(g);
        }
    }
}