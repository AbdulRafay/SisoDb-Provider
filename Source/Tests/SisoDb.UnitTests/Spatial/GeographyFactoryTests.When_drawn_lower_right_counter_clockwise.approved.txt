POLYGON ((-122.37185668945311 37.815566589202959, -122.36464691162108 37.818414354930027, -122.36258697509764 37.821804408982409, -122.36876678466795 37.831160150246035, -122.37340164184569 37.832244797136333, -122.37769317626952 37.830753403551292, -122.3792381286621 37.826821403234213, -122.37185668945311 37.815566589202959))